containers/java/.devcontainer/base.Dockerfile

```diff
@@ -0,0 +1,46 @@
+# Update the VARIANT arg in devcontainer.json to pick a Java version: 11, 14
+ARG VARIANT=11
+FROM openjdk:${VARIANT}-jdk-buster
```

Comment:
The purpose of adding a base.Dockerfile and Dockerfile is a little confusing to me. I see the main difference is the base image, which one do you recommend the user to use?

Reply:
The base.Dockerfile is used to generate the image. The Dockerfile is the one that will end up in VS Code. I'll update `devcontainer.json` to point to Dockerfile once the image has been published.  People do come here and just grab the `.devcontainer` folder so I don't want to break that. 😄 

[Lots more detail here](https://github.com/microsoft/vscode-dev-containers/blob/master/build/README.md) if you are interested.

Reply:
I should also add that the key difference is that all of the additional tools, zsh, etc are built into this image along with nvm and yarn for web development.

I'd wondered about including a version of gradle or maven - but it seems like that would age quickly.  Node.js itself isn't in the image, but nvm is which makes installing an updated version really easy and quick.